src/pybind/mgr/rgw/module.py

```diff
@@ -0,0 +1,197 @@
+from ceph.rgw.types import RGWAMException, RGWAMEnvMgr
+from ceph.rgw.rgwam_core import EnvArgs, RGWAM
+
+
+class RGWAMOrchMgr(RGWAMEnvMgr):
+    def __init__(self, mgr):
+        self.mgr = mgr
+
+    def apply_rgw(self, svc_id, realm_name, zone_name, port = None):
+        spec = RGWSpec(service_id = svc_id,
+                       rgw_realm = realm_name,
+                       rgw_zone = zone_name,
+                       rgw_frontend_port = port)
+        completion = self.mgr.apply_rgw(spec)
+        orchestrator.raise_if_exception(completion)
+
+    def list_daemons(self, service_name, daemon_type = None, daemon_id = None, host = None, refresh = True):
+        completion = self.mgr.list_daemons(service_name,
+                                           daemon_type,
+                                           daemon_id=daemon_id,
+                                           host=host,
+                                           refresh=refresh)
+        return orchestrator.raise_if_exception(completion)
+
+
+class Module(orchestrator.OrchestratorClientMixin, MgrModule):
+    MODULE_OPTIONS = []
+
+    # These are "native" Ceph options that this module cares about.
+    NATIVE_OPTIONS = []
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        self.inited = False
+        self.lock = threading.Lock()
+        super(Module, self).__init__(*args, **kwargs)
+
+        # ensure config options members are initialized; see config_notify()
+        self.config_notify()
+
+        with self.lock:
+            self.inited = True
+            self.env = EnvArgs(RGWAMOrchMgr(self),
+                               str(self.get_ceph_conf_path()),
+                               f'mgr.{self.get_mgr_id()}',
+                               str(self.get_ceph_option('keyring')))
+
+        # set up some members to enable the serve() method and shutdown()
+        self.run = True
+        self.event = threading.Event()
+
+
+
+    def config_notify(self) -> None:
+        """
+        This method is called whenever one of our config options is changed.
+        """
+        # This is some boilerplate that stores MODULE_OPTIONS in a class
+        # member, so that, for instance, the 'emphatic' option is always
+        # available as 'self.emphatic'.
+        for opt in self.MODULE_OPTIONS:
+            setattr(self,
+                    opt['name'],
+                    self.get_module_option(opt['name']))
+            self.log.debug(' mgr option %s = %s',
+                           opt['name'], getattr(self, opt['name']))
+        # Do the same for the native options.
+        for opt in self.NATIVE_OPTIONS:
+            setattr(self,
+                    opt,
+                    self.get_ceph_option(opt))
+            self.log.debug(' native option %s = %s', opt, getattr(self, opt))
+
+
+    @CLICommand('rgw admin', perm='rw')
+    def _cmd_rgw_admin(self, params: Sequence[str]):
+        """rgw admin"""
+        run_cmd = [ './bin/radosgw-admin',
```

Comment:
There is a new mgr method ``tool_exec`` which does these args for you

Reply:
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/mgr_module.py#L1560

Reply:
@liewegas @sebastian-philipp looking at `tool_exec`, it doesn't seem to pass in `-c`, for `ceph.conf`, any reason why not?

Reply:
I suspect it just wasn't needed